api/app/Models/BaseModel.php

```diff
@@ -3,12 +3,37 @@
 use Illuminate\Database\Eloquent\Model;
 use Bosnadev\Database\Traits\UuidTrait;
 
-class BaseModel extends Model
+abstract class BaseModel extends Model
 {
     use UuidTrait;
 
     public $incrementing = false;
 
+    /**
+     * The accessors to append to the model's array form.
+     *
+     * @var array
+     */
+    protected $appends = ['self'];
+
+    /**
+     * Get the access route for the entity.
+     *
+     * @return string
+     */
+    abstract public function entityRoute();
+
+    /**
+     * Get the user's first name.
+     *
+     * @param  string  $value
+     * @return string
+     */
+    public function getSelfAttribute()
+    {
+        return url($this->entityRoute().'/'.$this->{$this->primaryKey});
+    }
+
     public static function getTableName()
```

Comment:
Is there a way we can test this? It's the only remaining line that is not covered by unit tests.